### PR TITLE
Expose the traceur moduleName option as a function

### PR DESCRIPTION
**Pull request by rochejul**
To resolve the issue https://github.com/sindresorhus/gulp-traceur/issues/54

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+.idea
```
Comment (sindresorhus) [inline .gitignore]: nope

Comment (rochejul) [inline .gitignore]: Ok, I will remove that

**Comment by sindresorhus** [comment]
As already commented in https://github.com/sindresorhus/gulp-traceur/issues/54#issuecomment-69727190. It needs to use the same `NodeCompiler` instance for all files.

**Comment by rochejul** [comment]
Hi,

Finally, I use only one node compiler instance and set as second argument the module name.

**Comment by sindresorhus** [comment]
Please follow the existing code style.

**Comment by sindresorhus** [comment]
// @pkozlowski-opensource

**Comment by pkozlowski-opensource** [comment]
@sindresorhus I can review / try it out tomorrow, no more free cycles today. But happy to see a PR :-)

**Comment by pkozlowski-opensource** [comment]
@rochejul @sindresorhus I've finally got to reviewing this one. The good news is that it works for my use-case although I'm slightly worried by _how_ it works: it seems that we are relaying on several "tricks" here:
* current implementation still leaves out `moduleName` as part of options and we relaying on the fact that traceur will convert it to '"default"'
* we are relaying on the fact that traceur gets source name as module name - but what we are doing here is really changing the `sourceName` argument - which I assume is used for more than module name (ex.: error reporting).

@sindresorhus what is the hard motivation behind "same NodeCompiler instance for all files."? Perf?
I'm starting to think that maybe it makes sense to push this feature upstream, to the traceur itself?

Also, if we decide to merge this one, I think that a test could be written in a simpler way, for example making a function that adds / remove some prefix / suffix instead of having gigantic switch.

So again, this works, but IMO the implementation is rather shaky... so I don't feel to well abut it ...

**Comment by pkozlowski-opensource** [comment]
After digging into it some more I can confirm that the `sourceName` argument is used in other parts of traceur, ex.: for preparing source map urls. So I'm really not that comfortable with manipulating this arg just for the same of providing module name...

**Comment by pkozlowski-opensource** [comment]
Bah, looking more into traceur, the approach proposed by @rochejul might be the best we've got... Given this it would be +1 from me to merge this one given that:
* commits are squashed
* impl is simplified
* test is simplified

**Comment by sindresorhus** [comment]
@pkozlowski-opensource This is why it needs to be the same instance: https://github.com/sindresorhus/gulp-traceur/issues/47

I would much prefer for this to be somehow fixed in Traceur itself. Traceur is known to break public APIs, I don't even want to think of how fragile it is to depend on its internal behaviour.

**Comment by sindresorhus** [comment]
Should be fixed in Traceur.

**Pull request closed without merging**